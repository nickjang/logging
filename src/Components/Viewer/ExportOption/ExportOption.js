import React, { Component } from 'react';
import PropTypes from 'prop-types';
import './ExportOption.css'

class ExportOption extends Component {
  render() {
    const id = this.props.option.toLowerCase().replace(' ', '-');
    return (
      <div class='group-row' role={this.props.role || null}>
        <input
          type={this.props.type}
          id={id}
          name={this.props.name}
          ref={this.props.ref || null}
          value={id}
          form={this.props.form || null}
          checked={this.props.selected.includes(id)}
          onChange={(e) => this.props.update(e.target.value)} />
        <label for={id}>{this.props.option}</label>
      </div>
    );
  };
}

ExportOption.defaultProps = {
  type:'radio',
  option: '',
  name: '',
  ref: {},
  form: '',
  selected: [],
  update: ()=>{},
  role: ''
}

ExportOption.propTypes = {
  type: PropTypes.string.isRequired,
  option: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  ref: PropTypes.object,
  form: PropTypes.string,
  selected: PropTypes.arrayOf(PropTypes.string),
  update: PropTypes.func,
  role: PropTypes.string
}

export default ExportOption;