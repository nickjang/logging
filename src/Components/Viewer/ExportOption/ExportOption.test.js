import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import ExportOption from './ExportOption';

describe(`ExportOption component`, () => {
  it('renders ExportOption by default', () => {
    const wrapper = shallow(<ExportOption />);
    expect(toJson(wrapper)).toMatchSnapshot()
  });
})