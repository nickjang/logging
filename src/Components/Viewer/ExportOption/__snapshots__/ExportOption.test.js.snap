// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExportOption component renders ExportOption by default 1`] = `
<div
  className="group-row"
  role={null}
>
  <input
    form={null}
    id=""
    name=""
    onChange={[Function]}
    type="radio"
    value=""
  />
  <label
    htmlFor=""
  />
</div>
`;
